File: reader/server/core/backend/src/main/java/com/richodemus/reader/backend/subscription/SubscriptionRepository.java
package com.richodemus.reader.backend.subscription;

import com.richodemus.reader.backend.model.Feed;
import com.richodemus.reader.dto.FeedId;
import com.richodemus.reader.dto.ItemId;
import com.richodemus.reader.dto.UserId;

import java.util.List;

public interface SubscriptionRepository {
    List<Feed> get(final UserId userId);

    List<ItemId> get(final UserId userId, final FeedId feedId);

    void subscribe(final UserId userId, final FeedId feedId);

    void markAsRead(final UserId userId, final FeedId feedId, final ItemId itemId);

    void markAsUnread(UserId userId, FeedId feedId, ItemId itemId);
}

File: reader/server/core/backend/src/main/kotlin/com/richodemus/reader/backend/subscription/SubscriptionRepository.kt
package com.richodemus.reader.backend.subscription

import com.richodemus.reader.backend.model.Feed
import com.richodemus.reader.dto.FeedId
import com.richodemus.reader.dto.ItemId
import com.richodemus.reader.dto.UserId

interface SubscriptionRepository {
    operator fun get(userId: UserId): List<Feed>
    operator fun get(userId: UserId, feedId: FeedId): List<ItemId>
    fun subscribe(userId: UserId, feedId: FeedId)
    fun markAsRead(userId: UserId, feedId: FeedId, itemId: ItemId)
    fun markAsUnread(userId: UserId, feedId: FeedId, itemId: ItemId)
}
